docs/docs/reference/state-representation.md

```diff
@@ -0,0 +1,1757 @@
+position and velocity of a flying drone, or the measurement of an accelerometer or force-torque sensor are all examples
+of instantaneous state variables. The [`CartesianState`](#cartesian-state) and [`JointState`](#joint-state) classes
+can represent such values in an internally consistent way while providing useful functions for conversions,
+transformations and other manipulations.
+
+State classes are designed to be extensible to support more abstract state variables or properties. Another type of
+state is a **parameter**, which this library defines as a container class to for generic named variables.
+
+The following sections describe the properties of the main state classes in the library.
+
+## State
+
+The `State` base class defines the following attributes common to all derived states:
+
+- name: the name associated with the state
+- type: the specific StateType of the object
+- timestamp: an internal timestamp which refers to the last modification
+
+### Name
+
+The name is used to label a specific state instance, and can be used to disambiguate multiple states or check their
+compatibility. The name can be accessed or modified with `get_name()` and `set_name()` respectively.
+
+### State Type
+
+A state can hold different data depending on its type. The available state types are defined by the
+`state_representation::StateType` enumeration. The type field is accessible using the `get_type()` method and allows
+introspection when working with state pointers. It is a readonly property, determined by the constructor implementation.
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+void check_state_type(const std::shared_ptr<state_representation::State>& state) {
+  if (state->get_type() == ...) {
+    // 
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+
+### Timestamp
+
+The timestamp of a state records when it was last modified. Any non-const operations or mutations,
+including copy constructions, will reset the timestamp.
+
+`get_age()` returns the time since last modification in seconds, while `is_deprecated(time_delay)` can be used to
+check the age of a state against a maximum time delay.
+
+### Emptiness
+
+In addition, all states have a concept of "emptiness". The `State` base class does not refer to any specific state
+variable, and therefore holds no data. Derived classes that implement state variables do hold data. If a state is empty,
+it indicates that the data is uninitialized or otherwise invalid. The `is_empty()` method can be used to check
+emptiness, or otherwise the state object can be evaluated directly.
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+state_representation::State state;
+
+if (state) {
+  // the state is not empty
+} else {
+  // the state is empty
+}
+```
+
+</TabItem>
+</Tabs>
+
+A state can be marked as "empty" by calling `reset()`. The state remains empty until any data is set.
+
+## Cartesian state
```

Comment:
Do we somehow have to make it clear that all these derive from State so everything that we say above (name, timestamp, etc) holds here too?